sources/core/Stride.Core.CompilerServices/Analyzers/STRDIAG004PropertyWithNoGetter.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Immutable;
+using Stride.Core.CompilerServices.Common;
+
+namespace Stride.Core.CompilerServices.Analyzers;
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class STRDIAG004PropertyWithNoGetter : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "STRDIAG004";
+    private const string Title = "Property with no Getter";
+    private const string MessageFormat = "The [DataMember] Attribute is applied to property '{0}' with an invalid getter accessibility level. Expected is a public/internal getter.";
+    private const string Category = "CompilerServices";
+
+    private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+        DiagnosticId,
+        Title,
+        MessageFormat,
+        Category,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.EnableConcurrentExecution();
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+        context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+    }
+    private static void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+    {
+        var dataMemberAttribute = WellKnownReferences.DataMemberAttribute(context.Compilation);
+        if (dataMemberAttribute is null)
+        {
+            return;
+        }
+
+        context.RegisterSymbolAction(symbolContext => AnalyzeProperty(symbolContext, dataMemberAttribute), SymbolKind.Property);
+    }
+    private static void AnalyzeProperty(SymbolAnalysisContext context,INamedTypeSymbol dataMemberAttribute)
+    {
+        var propertySymbol = (IPropertySymbol)context.Symbol;
+        if(!WellKnownReferences.HasAttribute(propertySymbol, dataMemberAttribute)) 
+            return;
+        if(propertySymbol.DeclaredAccessibility != Accessibility.Public && propertySymbol.DeclaredAccessibility != Accessibility.Internal) 
+            return;
+        if (propertySymbol.GetMethod is null)
+        {
+            DiagnosticsAnalyzerExtensions.ReportDiagnostics(Rule, context, dataMemberAttribute, propertySymbol);
+        }
+        else
+        {
+            var getterAccessibility = propertySymbol.GetMethod.DeclaredAccessibility;
+
+            if (getterAccessibility != Accessibility.Public && getterAccessibility != Accessibility.Internal)
```

Comment:
Can we extract the accessibility check into a static helper method or ideally an extension method `IsVisibleToSerializer()` this way if we make changes to it in the future we can easily update it in one spot.

Reply:
[4cce494](https://github.com/stride3d/stride/pull/1864/commits/4cce4945fc9dc089e5c5d17990d441dfc2f43880)

made it an extension methods and added the missing checks

Reply:
is an extension method now